pkg/tfbridge/detailed_diff.go

```diff
@@ -435,27 +489,30 @@ func (differ detailedDiffer) makeSetDiff(
 
 	removed, added := computeSetHashChanges(oldIdentities, newIdentities)
 
+	if len(removed) == 0 && len(added) == 0 {
+		return nil
+	}
+
 	// We need to match the new indices to the inputs to ensure that the identity of the
 	// elements is preserved - this is necessary since the planning process can reorder
 	// the elements.
-	addedInputs := differ.matchNewIndicesToInputs(path, added)
-
-	changes := buildChangesIndexMap(addedInputs, removed)
-	for index, hashes := range changes {
-		oldVal := resource.NewNullProperty()
-		if removedIndex, ok := removed[hashes.oldHash]; ok {
-			oldVal = oldList[removedIndex]
-		}
-		newVal := resource.NewNullProperty()
-		if addedIndex, ok := added[hashes.newHash]; ok {
-			newVal = newList[addedIndex]
-		}
-
-		elemDiff := differ.makePropDiff(path.Index(int(index)), oldVal, newVal)
-		for subKey, subDiff := range elemDiff {
-			diff[subKey] = subDiff
+	matchedInputIndices := differ.matchPlanElementsToInputs(path, added, newList)
+	if matchedInputIndices == nil || len(matchedInputIndices) != len(added) {
+		// If we can't match the elements to the inputs, we will return a diff against the whole set.
+		// This ensures we still display a diff to the user, even if the algorithm can't determine
```

Comment:
So I really would like to know why we're bailing in this way that certainly removes information from the user, when in fact we could have had more information exposed, is this the right call? 

The alternative I have in mind here I'll sketch out on a branch.

Reply:
The current approximation may be viable too. Need to think on this.